xdsl/transforms/convert_qref_to_qssa.py

```diff
@@ -0,0 +1,46 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin
+from xdsl.dialects.qref import QRefBase
+from xdsl.dialects.qssa import QssaBase
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class QRefToQssaRewritePattern(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: QRefBase, rewriter: PatternRewriter):
+        # Create replacement operation
+        new_op: QssaBase = op.ssa_op()
+        # For gates there are no results to replace
+        new_results = () if new_op.is_gate else new_op.results
```

Comment:
Couldn't this check be replaced with a check of "has the new_op results that are a qubit"?

Reply:
`is_gate` should be equivalent to "The qubit-typed operands match one-to-one with the qubit-typed results".

Reply:
This property could be removed and replaced by an actual check of this. Essentially the `gates` are the things we actually need to modify in the transform as they switch between having results and having no results depending on the dialect. Alloc and measure on the other hand are basically unchanged between dialects (apart from their types)